src/middlewared/middlewared/plugins/reporting/netdata/graphs.py

```diff
@@ -365,18 +365,19 @@ def get_title(self):
         return 'Disk Temperature {identifier}'
 
     async def build_context(self):
-        self.disk_mapping = {}
         all_charts = await self.all_charts()
-        for disk in (await self.middleware.run_in_thread(get_disks_for_temperature_reading)).values():
-            identifier = disk.id if disk.id.startswith('nvme') else disk.serial
-            for k in (identifier, identifier.replace('-', '_')):
-                if f'smart_log_smart.disktemp.{k}' in all_charts:
-                    self.disk_mapping[get_human_disk_name(disk.__dict__)] = k
-                    break
+        self.disk_mapping = {
```

Comment:
We don't need to call `disk.query` here and maintain a disk mapping. We just need to call `iterate_disks` every 5 mins.